openvdb/openvdb/tools/FastSweeping.h

```diff
@@ -1296,6 +1310,24 @@ struct FastSweeping<SdfGridT, ExtValueT>::SweepingKernel
         inline operator bool() const { return v < SdfValueT(1000); }
     };// NN
 
+    template<typename ExtT = ExtValueT, typename SdfT = SdfValueT, typename std::enable_if<std::is_same<ExtT, int>::value, int>::type = 0>
+    ExtT twoNghbr(const NN& d1, const NN& d2, const SdfT& /* w */, const ExtT& v1, const ExtT& v2) const { return d1.v < d2.v ? v1 : v2; }// int implementation
+
+    template<typename ExtT = ExtValueT, typename SdfT = SdfValueT, typename std::enable_if<!std::is_same<ExtT, int>::value, int>::type = 0>
+    ExtT twoNghbr(const NN& d1, const NN& d2, const SdfT& w, const ExtT& v1, const ExtT& v2) const { return ExtT(w*(d1.v*v1 + d2.v*v2)); }// non-int implementation
+
+    template<typename ExtT = ExtValueT, typename SdfT = SdfValueT, typename std::enable_if<std::is_same<ExtT, int>::value, int>::type = 0>
+    ExtT threeNghbr(const NN& d1, const NN& d2, const NN& d3, const SdfT& /* w */, const ExtT& v1, const ExtT& v2, const ExtT& v3) const {
+        ExtT tmp = d1.v < d2.v ? v1 : v2;
+        SdfT sdfTmp = d1.v < d2.v ? d1.v : d2.v;
+        return sdfTmp < d3.v ? tmp : v3;
+    }// int implementation
```

Comment:
one more comment (though just a suggestion). if d1, d2, and d3 and v1, v2, and v3 were components of Vec3 vector, say d and v, then a faster (since it's branchless) and more elegant solution would simply read as:
 
return v[math::MinIndex(d)];

Not sure if that's possible without constructing temporary Vec3's, which might be costly.